src/main/java/com/restaurante/services/CompanyServices.java

```diff
@@ -19,45 +20,61 @@ public class CompanyServices {
 
     private final CompanyRepository companyRepository;
 
-    public ResponseEntity<Object> createCompany(CompanyDto companyDto) {
-        List<Company> companyList = companyRepository.findAll();
-        if(!companyList.isEmpty()){
-            //throw new CompanyExistException();
-            return ResponseEntity.status(HttpStatus.CONFLICT).body("There is already a company registered.");
+    public Company createCompany(CompanyDto companyDto) {
+        try {
+            List<Company> companyList = companyRepository.findAll();
+            if (!companyList.isEmpty()) {
+                throw new CompanyExistException();
+            }
+            Company company = new Company();
+            BeanUtils.copyProperties(companyDto, company);
+            companyRepository.save(company);
+            return null;
+        }catch (DataAccessException err){
+            throw new InternalServerErrorException();
         }
-        Company company = new Company();
-        BeanUtils.copyProperties(companyDto, company);
-        companyRepository.save(company);
-        return ResponseEntity.status(HttpStatus.CREATED).body("Company created.");
     }
 
-    public ResponseEntity<Object> listCompany(){
-        List<Company> companyList = companyRepository.findAll();
-        if(companyList.isEmpty()){
-            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Company not found.");
+    public List<CompanyDto> listCompany() {
+        try {
+            List<Company> companyList = companyRepository.findAll();
+            if (companyList.isEmpty()) {
+                throw new CompanyNotFoundException();
+            }
+            return (companyList.stream().map(CompanyDto::new).collect(Collectors.toList()));
+        }catch (DataAccessException err){
+            throw new InternalServerErrorException();
         }
-        return ResponseEntity.status(HttpStatus.OK).body(companyList.stream().map(CompanyDto::new).collect(Collectors.toList()));
     }
 
-    public ResponseEntity<Object> updateCompany(CompanyDto companyDto){
-        List<Company> companyList = companyRepository.findAll();
-        if(companyList.isEmpty()){
-            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Company not found.");
+    public Company updateCompany(CompanyDto companyDto){
+        try {
+            List<Company> companyList = companyRepository.findAll();
```

Comment:
Não seria mais vantajoso fazer uma busca por Id para verificar se existe a compania ?, poderia utilizar o metodo findById e varia uma busca mais rapida e deixaria mais clean o codigo
Exemplo:
```suggestion
            Company company = companyRepository.findById(companyDto.getId).orElseThrow(() -> new CompanyNotFoundException();
```

Reply:
Neste caso, se a pesquisa for vazia, ele entra no Throw. Para mim, deve ser ao contrário. Erro caso exista um cnpj.